### PR TITLE
feat: add support for derived distributions and gnome terminal themes

### LINKED ISSUE #81: update-sln: script is not posix compliant despite being in the default shell scripts
The `update-sln` script in the default shell (sh) script location is not posix compliant. It relies on arrays, which are not available in posix-compliant shells like dash.

### PR DESCRIPTION
**Pull request by dmccaffery**
- fix(template): do not assume dash is the default shell in sh profile
- fix: use posix-compliant source (dot) and exit codes
- fix: handle null globs correctly
- feat: add support for derived platforms (like popos)
- chore: normalise return values from functions
- chore: remove unnecessary vars from  all profiles
- fix(install-minikube): ensure minikube is installed via brew
- feat: add support for themes in gnome terminal

### CODE REVIEW

**Comment by dmccaffery** [comment]
derived install tested on Pop! OS: 
image

File: src/zsh/scripts/eval/set-completion [hunk 1]
```diff
@@ -38,7 +38,7 @@ if [ ! -d "$AM_PROMPT/zsh/completions" ]; then
 fi
 
 # iterate over each completion
-for AM_COMPLETION in $AM_PROMPT/zsh/completions/* 2>/dev/null; do
+for AM_COMPLETION in $AM_PROMPT/zsh/completions/*; do
```
Comment (patrickserrano) [inline src/zsh/scripts/eval/set-completion]: Oh nice! I noticed an issue with zsh completions today and was going to look into it when I had a minute.

**Review by patrickserrano — APPROVED** [review]
@dmccaffery looks like there are still some CI issues :/

I cloned and installed your branch locally and the issues I was having with zsh completions are resolved. So far I haven't run into any other issues which is a positive.

Cloning and installing this via `install.sh` did make me wonder if there's value in having a flag that would allow you to pass a GitHub url, so you could test installing via the bootstrap script? Might also help with CI since we were calling install directly there too, right?

**Comment by dmccaffery** [comment]
Yeah; something changed in how WSL is implemented in GitHub actions; gosu no longer persists through sudo, so homebrew installer is chowning the path as root:build with u=rw,go=r -- and the clone of the homebrew repo fails.

I'll be looking into it when I have a chance.

**Comment by dmccaffery** [comment]
> @dmccaffery looks like there are still some CI issues :/
> 
> I cloned and installed your branch locally and the issues I was having with zsh completions are resolved. So far I haven't run into any other issues which is a positive.
> 
> Cloning and installing this via `install.sh` did make me wonder if there's value in having a flag that would allow you to pass a GitHub url, so you could test installing via the bootstrap script? Might also help with CI since we were calling install directly there too, right?

re: passing in a git-url -- I'm not sure I understand the ask -- do you mean to run the bootstrap script against an arbitrary url, including forks? -- if so, that is incredibly dangerous -- its a huge security risk as one could redirect the bootstrap script to do whatever it wants, even from ci/cd on a fork -- the bootstrap script also calls install directly; all that it actually does is to extract a tarball of the repo into a temp directory and then invoke the install script. The real work achieved by the bootstrap script is to download the tarball... doing that against an arbitrary repo isn't helpful, I don't think.

**Comment by dmccaffery** [comment]
@patrickserrano : this is all that bootstrapping does:

https://github.com/automotiveMastermind/prompt/blob/d1eaccb93a7004ff382e94f2ca72b82749adb3b1/bootstrap.sh#L87-L90

The rest is just a comparison of the git-ref against the current upstream.

**Comment by dmccaffery** [comment]
@patrickserrano  / @sjk07 : should be ready for review / approval -- all builds should now pass. I had to do some gymnastics with the end-to-end build to get wsl to work again.

**Comment by patrickserrano** [comment]
> > @dmccaffery looks like there are still some CI issues :/
> > I cloned and installed your branch locally and the issues I was having with zsh completions are resolved. So far I haven't run into any other issues which is a positive.
> > Cloning and installing this via `install.sh` did make me wonder if there's value in having a flag that would allow you to pass a GitHub url, so you could test installing via the bootstrap script? Might also help with CI since we were calling install directly there too, right?
> 
> re: passing in a git-url -- I'm not sure I understand the ask -- do you mean to run the bootstrap script against an arbitrary url, including forks? -- if so, that is incredibly dangerous -- its a huge security risk as one could redirect the bootstrap script to do whatever it wants, even from ci/cd on a fork -- the bootstrap script also calls install directly; all that it actually does is to extract a tarball of the repo into a temp directory and then invoke the install script. The real work achieved by the bootstrap script is to download the tarball... doing that against an arbitrary repo isn't helpful, I don't think.

I hadn't really considered the security implications of such a change sine I was thinking of ease of manual testing larger PRs. But it makes sense why we wouldn't want to do that.

**Review by sjk07 — APPROVED** [review]
This is awesome stuff =]

**Pull request merged**